src/main/java/de/php_perfect/intellij/ddev/error_reporting/SentryErrorReporter.java

```diff
@@ -0,0 +1,112 @@
+    }
+
+    @Override
+    public boolean submit(IdeaLoggingEvent @NotNull [] events, @Nullable String additionalInfo, @NotNull Component parentComponent, @NotNull Consumer<? super SubmittedReportInfo> consumer) {
+        DataContext context = DataManager.getInstance().getDataContext(parentComponent);
+        Project project = CommonDataKeys.PROJECT.getData(context);
+
+        new Task.Backgroundable(project, "Sending error report") {
+            @Override
+            public void run(@NotNull ProgressIndicator indicator) {
+                List<SentryId> sentryIds = new ArrayList<>();
+                ThreadSync threadSync = new ThreadSync();
+
+                AtomicBoolean oneOrMoreFailed = new AtomicBoolean(false);
+
+                for (int i = 0; i < events.length; ++i) {
+                    SentryId sentryId = captureIdeaLoggingEvent(events[i], threadSync, oneOrMoreFailed);
+                    sentryIds.add(sentryId);
+
+                    indicator.setFraction((i + 1.0) / events.length);
+                }
+
+                announceReportSent(sentryIds, parentComponent);
+
+                SubmittedReportInfo.SubmissionStatus submissionStatus = oneOrMoreFailed.get() ?
+                        SubmittedReportInfo.SubmissionStatus.FAILED :
+                        SubmittedReportInfo.SubmissionStatus.NEW_ISSUE;
+                consumer.consume(new SubmittedReportInfo(submissionStatus));
+            }
+        }.queue();
+
+        return true;
+    }
+
+    private SentryId captureIdeaLoggingEvent(IdeaLoggingEvent event, ThreadSync threadSync, AtomicBoolean oneOrMoreFailed) {
+        SentryId sentryId = Sentry.captureEvent(buildSentryEvent(event), new SentrySubmissionResult((success) -> {
+            if (!success) {
+                oneOrMoreFailed.set(true);
+            }
+
+            threadSync.release();
+        }));
+
+        threadSync.waitForRelease();
+
+        return sentryId;
+    }
+
+    private SentryEvent buildSentryEvent(IdeaLoggingEvent ideaLoggingEvent) {
+        SentryEvent event = new SentryEvent();
+
+        event.setThrowable(ideaLoggingEvent.getThrowable());
+        if (ideaLoggingEvent instanceof IdeaReportingEvent && ideaLoggingEvent.getData() instanceof AbstractMessage) {
+            event.setThrowable(((AbstractMessage) ideaLoggingEvent.getData()).getThrowable());
+        }
+
+        event.setExtra("last_action_id", IdeaLogger.ourLastActionId);
+
+        if (getPluginDescriptor() instanceof IdeaPluginDescriptor) {
+            event.setRelease(getPluginDescriptor().getVersion());
+        }
+
+        return event;
+    }
+
+    private void announceReportSent(List<SentryId> sentryIds, Component parentComponent) {
+        String sentryIdsString = sentryIds.stream()
+                .map(SentryId::toString)
+                .collect(Collectors.joining(System.lineSeparator()));
+
+        ApplicationManager.getApplication().invokeLater(() -> Messages.showInfoMessage(
```

Comment:
Check out DdevNotifierImpl how to show notifications.

https://github.com/php-perfect/ddev-intellij-plugin/blob/509e09034b3538927616eca31acf0b3e7c4a1b3d/src/main/java/de/php_perfect/intellij/ddev/notification/DdevNotifierImpl.java#L113

Reply:
I'm not sure how I'd get the current project from the error reporting context for the Notifier. Do you have an idea for that?

Reply:
You already have access to the project, check out your call to TaskBackgroundable! ;)

Reply:
🤦🏻